assets/src/edit-story/components/library/panes/pageTemplates/savedTemplates.js

```diff
@@ -61,6 +70,28 @@ function SavedTemplates({
     setShowDialog(false);
   }, []);
 
+  const fetchTemplates = useCallback(() => {
+    if (!nextTemplatesToFetch) {
+      return;
+    }
+    getCustomPageTemplates(nextTemplatesToFetch)
+      .then(({ templates, hasMore }) => {
+        setSavedTemplates([...savedTemplates, ...templates]);
+        if (!hasMore) {
+          setNextTemplatesToFetch(false);
```

Comment:
Not a big fan of this happening here too. This feels like logic that should be moved up a step, no?